### PR TITLE
Time series updates for sharding support and updates / deletes

### PR DESCRIPTION
**Pull request by jeff-allen-mongo**
This update covers the changes in:

- [DOCSP-18792: Minimal support for sharding Time Series Collections](https://jira.mongodb.org/browse/DOCSP-18792)
- [DOCSP-16551: Time-Series Collection support for standard delete and update commands](https://jira.mongodb.org/browse/DOCSP-16551)

Staging links:

- [Time Series Sharding Limitations](https://docs-mongodbcom-staging.corp.mongodb.com/docs/docsworker-xlarge/time-series-updates/core/timeseries/timeseries-limitations/#sharding)
- [Time Series Granularity](https://docs-mongodbcom-staging.corp.mongodb.com/docs/docsworker-xlarge/time-series-updates/core/timeseries/timeseries-granularity/#change-the-granularity-of-a-time-series-collection)
- [Time Series Updates and Deletes Limitations](https://docs-mongodbcom-staging.corp.mongodb.com/docs/docsworker-xlarge/time-series-updates/core/timeseries/timeseries-limitations/#updates-and-deletes)
- [5.1 Release Notes](https://docs-mongodbcom-staging.corp.mongodb.com/docs/docsworker-xlarge/time-series-updates/release-notes/5.1/#time-series-collections)
- [shardCollection command](https://docs-mongodbcom-staging.corp.mongodb.com/docs/docsworker-xlarge/time-series-updates/reference/command/shardCollection/)

### CODE REVIEW

File: source/core/timeseries/timeseries-limitations.txt [hunk 1]
```diff
@@ -117,11 +126,14 @@ Client-Side Field Level Encryption
...
 :ref:`time series collections <manual-timeseries-collection>`.
 
+.. _time-series-limitations-sharding:
+
 Sharding
 ~~~~~~~~
 
-:ref:`Time series collections <manual-timeseries-collection>` cannot
-currently be sharded.
+Starting in MongoDB 5.1, sharded time series collections are supported.
```
Comment (banarun) [inline source/core/timeseries/timeseries-limitations.txt:134]: We should probably add more details here. Some of the key things to highlight would be,

1. The changes the the `shardCollection` command, which now accepts `timeseries` parameter.
2. The sharding admin commands (like split, moveChunk) are not supported for time-series collection. 
3. The limitations on the shard key pattern (Can only use `metaField` and `timeField` in the shard key).

**Review by banarun — COMMENTED** [review]
Thank you for adding the docs. I just added some suggestions for details for sharding time-series collection.

File: source/includes/time-series/fact-time-field-description.rst [hunk 2]
```diff
@@ -0,0 +1,3 @@
+Required. The name of the field which contains the date in each
+time series document. Documents in a time series collection must
+have a valid BSON date as the value for the ``timeField``.
```
Comment (jeff-allen-mongo) [inline source/includes/time-series/fact-time-field-description.rst:3]: Same as existing copy; shouldn't need review.

File: source/includes/time-series/fact-meta-field-description.rst [hunk 3]
```diff
@@ -0,0 +1,7 @@
+Optional. The name of the field which contains metadata in each
+time series document. The metadata in the specified field should
+be data that is used to label a unique series of documents. The
+metadata should rarely, if ever, change.
+
+The name of the specified field may not be ``_id`` or the same as
+the ``timeseries.timeField``. The field can be of any type.
```
Comment (jeff-allen-mongo) [inline source/includes/time-series/fact-meta-field-description.rst:7]: Same as existing copy; shouldn't need review.

File: source/includes/time-series/fact-granularity-description.rst [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+measurements.
+
+If you specify the ``timeseries.metaField``, consider the time
+span between consecutive incoming measurements that have the same
+unique value for the ``metaField`` field. Measurements often have
+the same unique value for the ``metaField`` field if they come
+from the same source.
+
+If you do not specify ``timeseries.metaField``, consider the time
+span between all measurements that are inserted in the collection.
```
Comment (jeff-allen-mongo) [inline source/includes/time-series/fact-granularity-description.rst:23]: Same as existing copy; shouldn't need review.

**Review by jeff-allen-mongo — COMMENTED** [review]
Thank you for your feedback @banarun. I've updated to include additional detail per your suggestions. Could you please take a look?

Cheers

File: source/reference/command/shardCollection.txt [hunk 5]
```diff
@@ -194,6 +253,20 @@ avoid scalability and perfomance issues.
   - :ref:`sharding-shard-key-selection`
   - :ref:`sharding-shard-key`
 
+Shard Keys on Time Series Collections
+`````````````````````````````````````
+
+When sharding time series collections, you can only specify the
+``metaField``, ``timeField``, or both in the shard key. No other fields,
```
Comment (banarun) [inline source/reference/command/shardCollection.txt:260]: We can also add `metaField (or sub-fields of a metaField)` to clarify that the field need not be just the `metaField`?

File: source/reference/command/shardCollection.txt [hunk 6]
```diff
@@ -194,6 +253,20 @@ avoid scalability and perfomance issues.
...
+including ``_id``, are allowed in the shard key pattern.
+
+- ``metaField`` can be either a :ref:`hashed shard key
+  <sharding-hashed-sharding>` or a :ref:`ranged shard key
+  <sharding-ranged>`.
+
+- ``timeField`` can only be a :ref:`ranged shard key
```
Comment (banarun) [inline source/reference/command/shardCollection.txt:267]: The `timeField` can only be at the end of the shard key pattern.

**Review by banarun — APPROVED** [review]
LGTM! Just a couple of minor comments. 

Are we using this ticket to also track the upgrade/downgrade produce? If yes, I think there are some FCV implications that we need to mention in the docs.

**Pull request merged**